x-pack/filebeat/input/awss3/sqs_s3_event.go

```diff
@@ -312,11 +316,13 @@ func (p *sqsS3EventProcessor) processS3Events(ctx context.Context, log *logp.Log
 	defer acker.Wait()
 
 	var errs []error
+	var handles []s3ObjectHandler
 	for i, event := range s3Events {
 		s3Processor := p.s3ObjectHandler.Create(ctx, log, client, acker, event)
 		if s3Processor == nil {
 			continue
 		}
+		handles = append(handles, s3Processor)
```

Comment:
this should go to https://github.com/elastic/beats/pull/33559/files#diff-082bcddae009852f8bbcc8844f783313e386164b88ee0317e36489bf340e2909R332
```
} else {
    handles = append(handles, s3Processor)
}
```

and when we return https://github.com/elastic/beats/pull/33559/files#diff-082bcddae009852f8bbcc8844f783313e386164b88ee0317e36489bf340e2909R335 we should return `handles` and not `nil` only if `len(handles) == len(s3Events)`

this way we are sure we backup and delete only if all the s3 objects in the sqs message were processed without error